components/properties/src/bidi.rs

```diff
@@ -0,0 +1,81 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::props::BidiClass;
+use icu_codepointtrie::CodePointTrie;
+use unicode_bidi::data_source::BidiDataSource;
+use unicode_bidi::BidiClass as DataSourceBidiClass;
+
+pub struct BidiClassAdapter<'a> {
```

Comment:
Nit: Everything public (exported) should have docs.

Reply:
Please:

1. On the struct `BidiClassAdapter`, add a docstest showing how to create one of these and then some methods you can call on it.
2. On the module `bidi` (in `//!` comments -- see other modules for examples), add a full-stack example demonstrating how to use this with the `unicode-bidi` crate to get the directionalities of an example sentence.

Reply:
done